### PR TITLE
Update to semconv 1.22

### PR DESCRIPTION
**Pull request by jack-berg**
Replaces #27.

### CODE REVIEW

**Comment by jack-berg** [comment]
I've added `japcmp` to this project to ensure we don't miss anything when manually adding any attributes that have been removed.

File: buildscripts/templates/SemanticAttributes.java.j2 [hunk 1]
```diff
@@ -667,6 +667,134 @@ public final class {{class}} {
...
+   *
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#JVM_MEMORY_TYPE} instead.
+   */
+  @Deprecated
+  public static final AttributeKey<String> TYPE = stringKey("type");
+
+  /**
+   * Physical server port.
+   *
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_LOCAL_PORT} instead.
```
Comment (trask) [inline buildscripts/templates/SemanticAttributes.java.j2:770]:
```suggestion
   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_LOCAL_PORT} on server telemetry and {@link SemanticAttributes#NETWORK_PEER_PORT} on client telemetry instead.
```

File: buildscripts/templates/SemanticAttributes.java.j2 [hunk 2]
```diff
@@ -667,6 +667,134 @@ public final class {{class}} {
...
+   */
+  @Deprecated
+  public static final AttributeKey<Long> SERVER_SOCKET_PORT = longKey("server.socket.port");
+
+  /**
+   * Immediate client peer address - unix domain socket name, IPv4 or IPv6 address.
+   *
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_PEER_ADDRESS} instead.
```
Comment (trask) [inline buildscripts/templates/SemanticAttributes.java.j2:778]:
```suggestion
   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_PEER_ADDRESS} on server telemetry and {@link SemanticAttributes#NETWORK_LOCAL_ADDRESS} on client telemetry instead.
```

File: buildscripts/templates/SemanticAttributes.java.j2 [hunk 3]
```diff
@@ -667,6 +667,134 @@ public final class {{class}} {
...
+   * The domain name of an immediate peer.
+   *
+   * <p>Notes:
+   *
+   * <ul>
+   *   <li>Typically observed from the client side, and represents a proxy or other intermediary
+   *       domain name.
+   * </ul>
+   *
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions.
```
Comment (trask) [inline buildscripts/templates/SemanticAttributes.java.j2:754]: (there's no replacement at this time)
```suggestion
   * @deprecated This item has removed in 1.22.0 of the semantic conventions.
```

File: buildscripts/templates/SemanticAttributes.java.j2 [hunk 4]
```diff
@@ -667,6 +667,134 @@ public final class {{class}} {
     private NetHostConnectionSubtypeValues() {}
   }
 
+  /**
+   * Immediate client peer port number.
+   *
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_PEER_PORT} instead.
```
Comment (trask) [inline buildscripts/templates/SemanticAttributes.java.j2:673]:
```suggestion
   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_PEER_PORT} on server telemetry and {@link SemanticAttributes#NETWORK_LOCAL_PORT} on client telemetry instead.
```

File: buildscripts/templates/SemanticAttributes.java.j2 [hunk 5]
```diff
@@ -667,6 +667,134 @@ public final class {{class}} {
...
+  /**
+   * The domain name of the source system.
+   *
+   * <p>Notes:
+   *
+   * <ul>
+   *   <li>This value may be a host name, a fully qualified domain name, or another host naming
+   *       format.
+   * </ul>
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions.
```
Comment (trask) [inline buildscripts/templates/SemanticAttributes.java.j2:701]:
```suggestion
   * @deprecated This item has removed in 1.22.0 of the semantic conventions.
```

File: buildscripts/templates/SemanticAttributes.java.j2 [hunk 6]
```diff
@@ -667,6 +667,134 @@ public final class {{class}} {
...
+   */
+  @Deprecated
+  public static final AttributeKey<String> SOURCE_DOMAIN = stringKey("source.domain");
+
+  /**
+   * Physical server IP address or Unix socket address. If set from the client, should simply use
+   * the socket's peer address, and not attempt to find any actual server IP (i.e., if set from
+   * client, this may represent some proxy server instead of the logical server).
+   *
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_LOCAL_ADDRESS} instead.
```
Comment (trask) [inline buildscripts/templates/SemanticAttributes.java.j2:711]:
```suggestion
   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_LOCAL_ADDRESS} on server telemetry and {@link SemanticAttributes#NETWORK_PEER_ADDRESS} on client telemetry instead.
```

File: buildscripts/templates/SemanticAttributes.java.j2 [hunk 7]
```diff
@@ -667,6 +667,134 @@ public final class {{class}} {
...
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#MESSAGING_MESSAGE_BODY_SIZE} instead.
+   */
+  @Deprecated
+  public static final AttributeKey<Long> MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES =
+      longKey("messaging.message.payload_size_bytes");
+
+  /**
+   * The domain name of the destination system.
+   *
+   * @deprecated This item has renamed in 1.22.0 of the semantic conventions.
```
Comment (trask) [inline buildscripts/templates/SemanticAttributes.java.j2:730]:
```suggestion
   * @deprecated This item has removed in 1.22.0 of the semantic conventions.
```

**Review by trask — APPROVED** [review]
❤️

**Pull request merged**